subcell_pipeline/analysis/dimensionality_reduction/pca_dim_reduction.py

```diff
@@ -36,11 +37,34 @@ def run_pca(data: pd.DataFrame) -> tuple[pd.DataFrame, PCA]:
         [pd.DataFrame(transform, columns=["PCA1", "PCA2"]), all_features],
         axis=1,
     )
-    pca_results = pca_results.sample(frac=1, random_state=1)
 
     return pca_results, pca
 
 
+def save_pca_results(
+    pca_results: pd.DataFrame, save_location: str, save_key: str, resample: bool = True
+) -> None:
+    """
+    Save PCA results data.
+
+    Parameters
+    ----------
+    pca_results
+        PCA trajectory data.
+    save_location
+        Location for output file (local path or S3 bucket).
+    save_key
+        Name key for output file.
+    resample : bool
+        True is data should be resampled before saving, False otherwise.
+    """
+
+    if resample:
```

Comment:
Q: What is the purpose of the resampling step?

Reply:
Scatter plot points are drawn in the order they appear in the data. By resampling, we "mix" the order that individual points show up to avoid an artifacts due to drawing order.

Reply:
Is this something that could be done at the plotting stage? It would be prefereable to have correspondence between saved PCA data and the saved fiber metrics. Another option could be to add a "resampled" suffix to help distinguish from non-shuffled data.

Reply:
It's faster to have it pre-shuffled for interactive plotting purposes, so I'd want to leave the option. Thoughts on just switching the default to unshuffled? The files that get saved to `analysis_outputs` are intended to be temporary, so I can just turn on the shuffling when saving for figure purposes

Reply:
> Thoughts on just switching the default to unshuffled? The files that get saved to `analysis_outputs` are intended to be temporary, so I can just turn on the shuffling when saving for figure purposes

If the output is meant to be temporary and it's faster to have it preshuffled, then I am okay with just going ahead with the current configuration! Just adding some explanation of the resampling step in the documentation would be helpful. Thanks!!